File: data/StarCoder2/readable_format_HPEC/fortran/file002.f90
program DRB024_simdtruedep_orig_yes
    use omp_lib
    implicit none

    integer :: i, len
    integer, dimension(:), allocatable :: a
    integer, dimension(:), allocatable :: b

    len = 100

    allocate (a(len))
    allocate (b(len))

    do i = 1, len
        a(i) = i
        b(i) = i+1
    end do

    !$omp simd
    do i = 1, len-1
        a(i+1) = a(i) + b(i)
    end do

    do i = 1, len
        write(6,*) 'Values for i and a(i) are:', i, a(i)
    end do

    deallocate(a,b)
end program

File: data/gpt-4o/readable_format_HPEC/cpp/file002.cpp
#include <stdio.h>
int main()
{
 int i, len = 100;
 int a[len], b[len];
 for (i=0; i<len; i++){
 a[i] = i+1;
 b[i] = i+2;
 }
#pragma omp simd
 for (i=0; i<len-1; i++)
 a[i+1] = a[i] + b[i];
 for (i=0; i<len; i++)
 printf("Values for i and a[i] are: %d %d\n", i+1, a[i]);
 return 0;
}
